src/main/java/seedu/address/model/profile/Profile.java

```diff
@@ -61,44 +55,43 @@ public Set<Tag> getTags() {
     }
 
     /**
-     * Returns true if both persons have the same name.
-     * This defines a weaker notion of equality between two persons.
+     * Returns true if both profiles have the same name.
+     * This defines a weaker notion of equality between two profiles.
      */
-    public boolean isSamePerson(Person otherPerson) {
-        if (otherPerson == this) {
+    public boolean isSameProfile(Profile otherProfile) {
+        if (otherProfile == this) {
             return true;
         }
 
-        return otherPerson != null
-                && otherPerson.getName().equals(getName());
+        return otherProfile != null
+                && otherProfile.getName().equals(getName());
```

Comment:
I think that since we are implementing nus emails, we should compare based on the email address. This should take care of the case sensitivity issue.

Comparing by email address seems better as it allows a user to have two profiles with the same name.